code/modules/hydroponics/seeds.dm

```diff
@@ -90,14 +90,23 @@
 			genes += new /datum/plant_gene/reagent(reag_id, reagents_add[reag_id])
 		reagents_from_genes() //quality coding
 
+/obj/item/seeds/Destroy()
+	// No AS ANYTHING here, because the list/genes could have typepaths in it.
+	for(var/datum/plant_gene/gene in genes)
+		if(genes.Remove(gene))
```

Comment:
I don't understand why you're doing this, this is going to be super expensive because `Remove` is O(n) (since it needs to preserve order.

You should instead:

```dm
for (var/datum/plant_gene/gene in genes)
    gene.on_removed(src)
    qdel(gene)

genes.Cut()
```

Reply:
I had no idea Remove operates like that 🥴

Reply:
I mean "super expensive"

Reply:
Iterating through the list is already O(n), removing through the list as you iterate it is O(n^2)!